### PR TITLE
Add maxWriteReplicationIndex and syncUntil in Database

### PR DESCRIPTION
**Pull request by haaawk**
`Database::maxWriteReplicationIndex` can be used by the client to keep track of the lowest replication index that guarantees seeing all the writes done by this client. `Database::syncUntil` can be then used to sync the embedded replica at least to the replication index given by the client.

That way a client can:

Do a bunch of operations on one embedded replica
Fetch the max write replication index I
Run syncUntil with a given replication index I on a different embedded replica and be sure that they see all the operations they've done on the first embedded replica.

### CODE REVIEW

File: types/promise.d.ts [hunk 1]
```diff
@@ -35,6 +36,7 @@ declare class Database {
     aggregate(name: any, options: any): void;
     table(name: any, factory: any): void;
     loadExtension(...args: any[]): void;
+    maxWriteReplicationIndex(): any;
```
Comment (sivukhin) [inline types/promise.d.ts:39]: Why `any` and not `number | undefined`?

**Review by sivukhin — APPROVED** [review]
Should we also bump version in package.json?

File: types/promise.d.ts (hunk 1 above)
Comment (haaawk) [reply types/promise.d.ts:39]: This is generated so you have to ask the generator :)

**Comment by haaawk** [comment]
> Should we also bump version in package.json?

done

**Pull request merged**